src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -187,16 +200,90 @@ class ContentOpenWith extends PureComponent<Props, State> {
             .getOpenWithIntegrations(fileId, language, this.fetchOpenWithSuccessHandler, this.fetchErrorHandler);
     }
 
+    setResolvedIntegrationState = (integrations: Array<Integration>) =>
+        this.setState({ integrations, isLoading: false });
+
     /**
      * Fetch app integrations info needed to render.
      *
      * @param {OpenWithIntegrations} integrations - The available Open With integrations
      * @return {void}
      */
-    fetchOpenWithSuccessHandler = (integrations: Array<Integration>): void => {
+    fetchOpenWithSuccessHandler = async (integrations: Array<Integration>): Promise<any> => {
+        const boxEditIntegration = integrations.find(({ appIntegrationId }) =>
+            this.isBoxEditIntegration(appIntegrationId),
+        );
+
+        if (!boxEditIntegration || boxEditIntegration.isDisabled) {
+            this.setState({ integrations, isLoading: false });
+            return;
+        }
+
+        try {
+            const { extension } = await this.getIntegrationFileExtension();
+            boxEditIntegration.extension = extension;
```

Comment:
It seems like you could create a new object to contain your overrides, then spread it onto the `boxEditIntegration` object at the end of the method, avoiding duplication of the `setState` call.